clusterloader2/pkg/measurement/common/scheduling_throughput.go

```diff
@@ -32,153 +29,86 @@ import (
+	summaryName, err := util.GetStringOrDefault(config.Params, "summaryName", a.String())
 	if err != nil {
-		return fmt.Errorf("pod store creation error: %v", err)
+		return nil, err
 	}
-	s.isRunning = true
-	klog.V(2).Infof("%s: starting collecting throughput data", s)
-
-	go func() {
-		defer ps.Stop()
-		lastScheduledCount := 0
-		for {
-			select {
-			case <-s.stopCh:
-				return
-			case <-time.After(measurmentInterval):
-				pods := ps.List()
-				podsStatus := measurementutil.ComputePodsStartupStatus(pods, 0, nil /* updatePodPredicate */)
-				throughput := float64(podsStatus.Scheduled-lastScheduledCount) / float64(measurmentInterval/time.Second)
-				s.schedulingThroughputs = append(s.schedulingThroughputs, throughput)
-				lastScheduledCount = podsStatus.Scheduled
-				klog.V(3).Infof("%v: %s: %d pods scheduled", s, selector.String(), lastScheduledCount)
-			}
-		}
-	}()
-	return nil
-}
 
-func (s *schedulingThroughputMeasurement) gather(threshold float64) ([]measurement.Summary, error) {
-	if !s.isRunning {
-		klog.Errorf("%s: measurement is not running", s)
-		return nil, fmt.Errorf("measurement is not running")
-	}
-	s.stop()
-	klog.V(2).Infof("%s: gathering data", s)
-
-	throughputSummary := &schedulingThroughput{}
-	if length := len(s.schedulingThroughputs); length > 0 {
-		sort.Float64s(s.schedulingThroughputs)
-		throughputSummary.Perc50 = s.schedulingThroughputs[int(math.Ceil(float64(length*50)/100))-1]
-		throughputSummary.Perc90 = s.schedulingThroughputs[int(math.Ceil(float64(length*90)/100))-1]
-		throughputSummary.Perc99 = s.schedulingThroughputs[int(math.Ceil(float64(length*99)/100))-1]
-		throughputSummary.Max = s.schedulingThroughputs[length-1]
-	}
 	content, err := util.PrettyPrintJSON(throughputSummary)
 	if err != nil {
 		return nil, err
 	}
-	summary := measurement.CreateSummary(schedulingThroughputMeasurementName, "json", content)
+
+	summaries := []measurement.Summary{
+		measurement.CreateSummary(summaryName, "json", content),
+	}
+
+	threshold, err := util.GetFloat64OrDefault(config.Params, "threshold", 0)
+
 	if threshold > 0 && throughputSummary.Max < threshold {
 		err = errors.NewMetricViolationError(
 			"scheduler throughput",
 			fmt.Sprintf("actual throughput %f lower than threshold %f", throughputSummary.Max, threshold))
+		klog.Errorf("%s: %v", a, err)
 	}
-	return []measurement.Summary{summary}, err
+
+	return summaries, err
 }
 
-func (s *schedulingThroughputMeasurement) stop() {
-	if s.isRunning {
-		close(s.stopCh)
-		s.isRunning = false
+func (a *schedulingThroughputGatherer) getThroughputSummary(executor QueryExecutor, startTime time.Time, config *measurement.Config) (*schedulingThroughput, error) {
+
+	measurementEnd := time.Now()
+	measurementDuration := measurementEnd.Sub(startTime)
+	promDuration := measurementutil.ToPrometheusTime(measurementDuration)
+
+	query := fmt.Sprintf(maxSchedulingThroughputQuery, promDuration)
+	samples, err := executor.Query(query, measurementEnd)
+
```

Comment:
nit: remove this empty line

Generally, you should split the conceptual blocks with empty lines - error handling is clearly the same block as where the err was computed.